src/Storages/ColumnsDescription.cpp

```diff
@@ -173,16 +173,14 @@ static auto getNameRange(const ColumnsDescription::ColumnsContainer & columns, c
 {
     String name_with_dot = name_without_dot + ".";
 
-    auto begin = columns.begin();
-    for (; begin != columns.end(); ++begin)
+    for (auto it = columns.begin(); it != columns.end(); ++it)
     {
-        if (begin->name == name_without_dot)
-            return std::make_pair(begin, std::next(begin));
-
-        if (startsWith(begin->name, name_with_dot))
-            break;
+        if (it->name == name_without_dot)
+            return std::make_pair(it, std::next(it));
     }
 
+    auto begin = std::find_if(columns.begin(), columns.end(), [&](const auto & column){ return startsWith(column.name, name_with_dot); });
```

Comment:
Maybe just return previous code, but assign `begin` before `break`?

Reply:
> but assign begin before break

What do you mean?
The problem was in that we can have column with name `name_without_dot` and columns that start with `name_with_dot`. So first we need to check if we have column with name `name_without_dot` and if not - check if we have names that start with `name_with_dot`. Previously we could stop searching name `name_without_dot` because we found name that starts with `name_with_dot`, and it caused bugs

Reply:
Clear to me now, but maybe add this to comments.